core/services/workflows/utils.go

```diff
@@ -0,0 +1,114 @@
+package workflows
+
+import (
+	"context"
+	"fmt"
+	"go.opentelemetry.io/otel/attribute"
+	"reflect"
+)
+
+const WorkflowID = "WorkflowID"
+const WorkflowExecutionID = "WorkflowExecutionID"
+
+type keystoneWorkflowContextKey struct{}
```

Comment:
Small note re naming: can we call this WorkflowContext, omitting Keystone from the naming? I think it's very verbose and given the location of this it's clearly anchored towards the workflow engine.